samples/AspNetIntegration/Program.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license informations
+
+using AspNetIntegration;
+using Microsoft.Extensions.Hosting;
+
+//<docsnippet_aspnet_registration>
+var host = new HostBuilder()
+    .ConfigureFunctionsWebApplication(builder =>
+    {
+        // can still register middleware and use this extension method the same way
+        // .ConfigureFunctionsWorkerDefaults() is used
+        builder.UseWhen<FooMiddleware>((context)=>
```

Comment:
This code can't be used in a doc snippet as it isn't self-contained (i.e., it contains a reference to a class outside of the snippet, `FooMiddleware`), so when shown in isolation, customers can't copy/paste and compile.